File: app/src/main/java/com/friday/ar/util/UpdateUtil.java
package com.friday.ar.util;

import android.content.Context;
import android.content.pm.PackageManager;
import android.util.Log;

import androidx.annotation.NonNull;

import com.crashlytics.android.Crashlytics;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class UpdateUtil {
    private static final String LOGTAG = "UpdateUtil";
    private static FirebaseDatabase database = FirebaseDatabase.getInstance();
    private static DatabaseReference updateRef = database.getReference("version");

    public static void checkForUpdate(Context context) {
        updateRef.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                try {
                    if (!context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName.equals(dataSnapshot.getValue().toString())) {
                        NotificationUtil.notifyUpdateAvailable(context, dataSnapshot.getValue().toString());
                    }
                } catch (PackageManager.NameNotFoundException e) {
                    Log.e(LOGTAG, e.getLocalizedMessage(), e);
                }
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
                Log.e(LOGTAG, "Could not check for update:" + databaseError.getMessage() + "\nDetails:" + databaseError.getDetails(), databaseError.toException());
                Crashlytics.logException(databaseError.toException());
            }
        });
    }
}

File: app/src/main/java/com/friday/ar/util/UpdateUtil.kt
package com.friday.ar.util

import android.content.Context
import android.content.pm.PackageManager
import android.preference.PreferenceManager
import android.util.Log
import com.crashlytics.android.Crashlytics
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

object UpdateUtil {
    private const val LOGTAG = "UpdateUtil"
    fun checkForUpdate(context: Context) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val database = FirebaseDatabase.getInstance()
        val updateRef = if (sharedPreferences.getBoolean("pref_devmode_use_beta_channel", false)) database.getReference("versionPre") else database.getReference("version")
        updateRef.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(dataSnapshot: DataSnapshot) {
                try {
                    if (context.packageManager.getPackageInfo(context.packageName, 0).versionName != dataSnapshot.value!!.toString()) {
                        NotificationUtil.notifyUpdateAvailable(context, dataSnapshot.value!!.toString())
                    }
                } catch (e: PackageManager.NameNotFoundException) {
                    Log.e(LOGTAG, e.localizedMessage, e)
                }

            }

            override fun onCancelled(databaseError: DatabaseError) {
                Log.e(LOGTAG, "Could not check for update:" + databaseError.message + "\nDetails:" + databaseError.details, databaseError.toException())
                Crashlytics.logException(databaseError.toException())
            }
        })
    }
}
